### ISSUE TITLE
2FA secrets sent to google

### ISSUE THREAD

**Issue by mtausig** [snipe/snipe-it#6402]
#### Expected Behavior (or desired behavior if a feature request)

If I am activating 2 factor authentication, the shared secret between the server and the authenticator app should only be known to these two parties.

-----

#### Actual Behavior

The QR code containing the secret is not calculated and sent by the server itself but displayed via a link to charts.googleapis.com. Thus every single 2FA secret is sent to a third party (Google) which is a security problem

-----

#### Please confirm you have done the following before posting your bug report:

- [ ] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? 

fresh

- Version of Snipe-IT you're running

v 4.5.0

- Version of PHP you're running

7

- What OS and web server you're running Snipe-IT on

nginx, Ubuntu 16.04

- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error

https://snipeit.myurl.com/two-factor-enroll

**Comment by EarlRamirez** [comment]
What steps should I use to reproduce this?

**Comment by mtausig** [comment]
1. Login to your account on a snipe-it instance
2. "Edit your Profile"
3. "Enable Two Factor"
4. "Save"
5. Right Click on the QR Code and select "Inspect Element"
6. Observe that the source of the code is `<img src="
https://chart.googleapis.com/chart?chs=...&chl=otpauth%3A%2F%2Ftotp%2FINSTANCENAME%2BInventory%3AUSERNAME%3Fsecret%3DTHEVALUEOFTHESECRETWHICHSHOULDREMAINSECRET%26issuer%3DINSTANCENAME
" style="...">`

On Mon, 2018-12-03 at 23:38 +0000, Earl Ramirez wrote:
> What steps should I use to reproduce this?
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
--
DI Mathias Tausig
Lehre und Forschung
Kompetenzzentrum für IT-Security

FH Campus Wien
Masterstudium IT-Security
Favoritenstrasse 226, Raum B.2.15
1100 Wien, Austria
T: +43 1 606 68 77-2472
F: +43 1 606 68 77-2469
mathias.tausig@fh-campuswien.ac.at
PGP Key-ID: 75656BBF
Informationen zum Datenschutz: www.fh-campuswien.ac.at/datenschutzerklaerung

**Comment by mtausig** [comment]
Yes. The *Inspect Element* Option is available (at least) in Firefox and Chromium.
As an alternative, you can simply open the source code of the website with the QR code and search for "charts.googlepais.com"

Informationen zum Datenschutz: www.fh-campuswien.ac.at/datenschutzerklaerung

**Comment by mtausig** [comment]
No, I am not talking about the QR code linking to an asset.
I am talking about the QR code used to initialise Two factor authentication.

Informationen zum Datenschutz: www.fh-campuswien.ac.at/datenschutzerklaerung

**Comment by EarlRamirez** [comment]
@mtausig I was able to confirm this

**Comment by snipe** [comment]
Hi @mtausig - Sorry for the delay, I didn't see this one - which is why we specifically request that security issues are emailed to us, not submitted via GH issue, in our Readme...

We're looking into the issue.

**Comment by snipe** [comment]
It’s literally in the readme. Go to to the main page for this project here on this website, scroll down. It’s been there for years. 

Your concern is valid, and that’s why we’re addressing it. But you might try being less shitty next time, when you don’t follow protocol and expect the world to bend around you.

**Comment by mtausig** [comment]
If "missing an information hidden beneath 6 pages of avatar pictures an politely suggesting a better placement for it" is shitty behaviour on your page, than we do have quite a different definition of that.

But I am glad, that you are on the issue now.

**Comment by snipe** [comment]
So, we work for years to give you a free product and scrolling down a page is too much to ask. Got it. 

The shittiness I spoke of is "Doesn't anybody here care for a security issue?". Of course we care. Github Issues is not ideal for a variety of reasons, and this is one of them. There is discussion on the maintainers GH list to provide a better way to discreetly file security issues so that they get higher priority in a never-ending sea of issues, but we're not there yet. Had you emailed, this would have been addressed a lot faster.

**Comment by mtausig** [comment]
1. I read the contributing documentation provided
2. I read the "common issues" and "getting help" documents referenced there
3. I read the extensive issue template

I proceeded as was suggested by those, since none of them stated anything specific about security issues. After a month has passed, I bumped the issue to prevent it from being forgotten. This is definitely the first project I have contributed to, where this is considered offensive behaviour.
